### ISSUE TITLE
Clean up use of unittest assert functions in twisted/python

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8389]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8389|
|Type|enhancement|
|Created|2016-06-03 02:13:41Z|
|Branch|https://github.com/twisted/twisted/tree/8389-rodrigc-assert|

This ticket is split out from [ticket:8371] because the patch there was too big to review.

The files in twisted/python
should have their use of unittest assert functions cleaned up so that:

* the most appropriate unittest assert function is used appropriately in each file
* use of deprecated aliases should be eliminated, to reduce deprecation
  warnings and errors when moving to Python3 compatibility

This has some pointers about deprecated unittest assert aliases:

See: https://docs.python.org/2.7/library/unittest.html#deprecated-aliases

|= Method Name =|= Deprecated alias(es) =|
|---------------|------------------------|
| assertEqual() | failUnlessEqual, assertEquals |
| assertNotEqual() | failIfEqual |
| assertTrue() | failUnless, assert_ |
| assertFalse() | failIf |
| assertRaises() |	failUnlessRaises |
| assertAlmostEqual() | failUnlessAlmostEqual  |
| assertNotAlmostEqual() | failIfAlmostEqual  |

<details><summary>Searchable metadata</summary>

```
trac-id__8389 8389
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8389_rodrigc_assert 8389-rodrigc-assert
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464920021964444 1464920021964444
changetime__1466614773244292 1466614773244292
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/105

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @rodrigc|
|-|-|

Thanks for your contribution, rodrigc.

As with [#8376](https://github.com/twisted/twisted/issues/8376), the changes to these assertion functions are over-broad.  These are a lot of changes to look over, but please re-submit more minimal changes that just address the necessary bits to migrate to python 3.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @wsanchez|
|-|-|

Thanks for helping with other reviews.
Can you look at this one?

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_5"></a>|@wsanchez set owner to @rodrigc|
|-|-|

reviewed.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset 4a4d7ae38be9894eb1b4cbdb56bd3f9e9d525f8e
```
#!CommitTicketReference repository="" revision="4a4d7ae38be9894eb1b4cbdb56bd3f9e9d525f8e"
Merge 8389-rodrigc-assert: Clean up use of unittest assert functions in twisted/python

Author: rodrigc
Reviewer: wsanchez
Fixes: #8389
```
